demisto_sdk/commands/download/downloader.py

```diff
@@ -750,520 +863,434 @@ def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
+        if not summary_log:
+            summary_log = "No files were downloaded."
+
+        logger.info(summary_log.rstrip("\n"))
+
+        return not failed_downloads_count  # Return True if no downloads failed, False otherwise.
 
-    def merge_and_extract_existing_file(self, custom_content_object: dict) -> None:
+    def download_unified_content(self, content_object: dict,
+                                 existing_pack_structure: dict[str, dict[str, list[dict]]],
+                                 output_path: Path,
+                                 overwrite_existing: bool = False) -> bool:
         """
-        "Smart" merges old files of type integration/script (existing in the output pack)
-        :param custom_content_object: The custom content object to merge into the pack
-        :return: None
+        Download unified content items.
+        Existing content items will be skipped if 'overwrite_existing' is False.
+        A "smart" merge will be done for pre-existing YAML files, adding fields that exist in existing file,
+        but were omitted by the server.
+
+        Args:
+            content_object (dict): The content object to download
+            existing_pack_structure (list[dict]): A list of existing content item files in the output pack.
+            output_path (Path): The output path to write the files to.
+            overwrite_existing (bool): Whether to overwrite existing files or not.
+
+        Returns:
+            bool: True if the content item was downloaded successfully, False otherwise.
         """
-        file_path: str = custom_content_object["path"]
-        file_name: str = custom_content_object["name"]
-        file_type: str = custom_content_object["type"]
-        file_entity: str = custom_content_object["entity"]
+        temp_dir: str | None = None
+        content_item_name: str = content_object["name"]
+        content_item_type: FileType = content_object["type"]
+        content_item_entity: str = content_object["entity"]
+        content_directory_name = self.create_directory_name(content_item_name)
+
+        content_item_exists = (  # Content item already exists in output pack
+            content_item_name in existing_pack_structure.get(content_item_entity, {})
+        )
 
-        file_code_language: str = custom_content_object.get("code_lang", "")
-        if not self.verify_code_lang(file_code_language, file_type, file_name):
-            return
+        if content_item_exists:
+            if not overwrite_existing:  # If file exists, and we don't want to overwrite it, skip it.
+                logger.debug(
+                    f"Content item '{content_item_name}' already exists in output pack. Skipping..."
+                )
+                return False
+
+            # If we overwrite existing files, we need to extract the existing files to a temp directory
+            # for a "smart" merge.
+            temp_dir = mkdtemp()
```

Comment:
This is problematic since the temp directory is generated only in-case the file content item already exists in the output path (otherwise it's just directly saved inside it without using a temp dir). If we add a `with` block, it'll require creating a temp dir every single time (even if it's not needed, and the file doesn't already exist.
See the full function for reference.